src/Uno.UI/UI/Xaml/DependencyObjectStore.Binder.cs

```diff
@@ -253,15 +253,13 @@ private void BinderDispose()
 				_isDisposed = true;
 			}
 
-			_properties.Dispose();
+			_properties?.Dispose();
 			_childrenBindableMap.Dispose();
 		}
 
 		private void OnDataContextChanged(object? providedDataContext, object? actualDataContext, DependencyPropertyValuePrecedences precedence)
 		{
-			var dataContextBinding = _properties.FindDataContextBinding();
-
-			if (dataContextBinding != null && precedence == DependencyPropertyValuePrecedences.Inheritance)
+			if (precedence == DependencyPropertyValuePrecedences.Inheritance && Properties.FindDataContextBinding() is { } dataContextBinding)
```

Comment:
Re-ordering the conditions to avoid doing `Properties.FindDataContextBinding()` unnecessarily.